src/Neo/SmartContract/Native/LedgerContract.cs

```diff
@@ -47,13 +47,21 @@ internal override ContractTask OnPersist(ApplicationEngine engine)
             engine.Snapshot.Add(CreateStorageKey(Prefix_Block).Add(engine.PersistingBlock.Hash), new StorageItem(Trim(engine.PersistingBlock).ToArray()));
             foreach (TransactionState tx in transactions)
             {
+                // It's possible that there are previously saved malicious conflict records for this transaction.
+                // These records are garbage in fact and can be removed in a separate flow, but at the same time
+                // it won't hurt to keep them.
+
                 engine.Snapshot.Add(CreateStorageKey(Prefix_Transaction).Add(tx.Transaction.Hash), new StorageItem(tx));
+
+                // Store transaction's conflicits.
                 var conflictingSigners = tx.Transaction.Signers.Select(s => s.Account);
                 foreach (var attr in tx.Transaction.GetAttributes<Conflicts>())
                 {
-                    var conflictRecord = engine.Snapshot.GetAndChange(CreateStorageKey(Prefix_Transaction).Add(attr.Hash),
-                        () => new StorageItem(new TransactionState { ConflictingSigners = Array.Empty<UInt160>() })).GetInteroperable<TransactionState>();
-                    conflictRecord.ConflictingSigners = conflictRecord.ConflictingSigners.Concat(conflictingSigners).Distinct().ToArray();
+                    engine.Snapshot.Add(CreateStorageKey(Prefix_Transaction).Add(attr.Hash), new StorageItem(new TransactionState() { BlockIndex = engine.PersistingBlock.Index }));
```

Comment:
I tried to hack a private net by the code from https://github.com/neo-project/neo/issues/2907, when multi transactions want to cancel the same tx(the same conflict attribute), it will throw `InvalidOperationException`  when they all change the same key's vale in the same block. So finally it will be like:
![image](https://github.com/neo-project/neo/assets/38493437/8d67c36a-bf48-4caf-99b2-cd45add4fe41)
![1699261532080](https://github.com/neo-project/neo/assets/38493437/377b1347-9c3b-49ef-abcb-a9866b7e0130)

Reply:
It's my poor knowledge of NeoC# codebase, the intention here was to replace the existing conflict record (if it's in the DB), but `engine.Snapshot.Add` doesn't allow to do it. I'll fix it, thank you for testing!

Reply:
@superboyiii, could you, please, run the same test? I've modified the code so that now it's possible to rewrite existing malicious conflict record with the proper transaction.